### PR TITLE
Partial merge of the grid layout infrastructure

### PR DESCRIPTION
**Pull request by Alotor**
This MR contains the code currently in development of the grid layout infrastructure. The functionality is disabled for the moment but it's a partial merge to minimize future conflicts.

### CODE REVIEW

File: frontend/src/app/main/data/workspace/shapes.cljs [hunk 1]
```diff
@@ -82,6 +82,7 @@
    (ptk/reify ::add-shape
      ptk/WatchEvent
      (watch [it state _]
+       (.log js/console (clj->js attrs))
```
Comment (superalex) [inline frontend/src/app/main/data/workspace/shapes.cljs]: Ups!

File: common/src/app/common/types/shape/layout.cljc [hunk 2]
```diff
@@ -408,3 +497,91 @@
...
+
+;; TODO: Mix the cells given as arguments leaving only one. It should move all the shapes in those cells in the direction for the grid
+;; and lastly use assign-cells to reassing the orphaned shapes
+(defn merge-cells
+  [parent _cells]
+  parent)
+
+
+;; TODO
+;; Assign cells takes the children and move them into the aloted cells. If there are not enough cells it creates
```
Comment (superalex) [inline common/src/app/common/types/shape/layout.cljc]: aloted?

File: frontend/src/app/main/data/workspace/shapes.cljs [hunk 3]
```diff
@@ -319,7 +322,6 @@
            (dc/detach-comment-thread ids)
            (ptk/data-event :layout/update all-parents)
            (dch/commit-changes changes)
-           (ptk/data-event :layout/update layout-ids)
```
Comment (superalex) [inline frontend/src/app/main/data/workspace/shapes.cljs:322]: Why this disappears?

File: frontend/src/app/main/data/workspace/transforms.cljs [hunk 4]
```diff
@@ -438,9 +439,10 @@
                    (cph/selected-with-children objects selected))
 
              exclude-frames-siblings
-             (into exclude-frames
+             exclude-frames
```
Comment (superalex) [inline frontend/src/app/main/data/workspace/transforms.cljs]: Why this changes?

File: frontend/src/app/main/data/workspace/shapes.cljs (hunk 3 above)
Comment (Alotor) [reply frontend/src/app/main/data/workspace/shapes.cljs:322]: The previous `:layout/update` should reflow the layout so the second is not necessary and it's better performance.

File: frontend/src/app/main/data/workspace/transforms.cljs (hunk 4 above)
Comment (Alotor) [reply frontend/src/app/main/data/workspace/transforms.cljs]: Shouldn't change. Thanks
